source/common/upstream/load_balancer_impl.h

```diff
@@ -542,16 +585,21 @@ class LeastRequestLoadBalancer : public EdfLoadBalancerBase,
     //
     // It might be possible to do better by picking two hosts off of the schedule, and selecting the
     // one with fewer active requests at the time of selection.
-    if (active_request_bias_ == 0.0) {
-      return host.weight();
-    }
+
+    double host_weight = static_cast<double>(host.weight());
 
     if (active_request_bias_ == 1.0) {
-      return static_cast<double>(host.weight()) / (host.stats().rq_active_.value() + 1);
+      host_weight = static_cast<double>(host.weight()) / (host.stats().rq_active_.value() + 1);
+    } else if (active_request_bias_ != 0.0 && active_request_bias_ != 1.0) {
```

Comment:
`active_request_bias_` can't be 1.0 here because it would have made the `if` statement true?